### PR TITLE
chore(docs): Use JSDoc @deprecated 

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
# ...

Attach a link to issue if relevant

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **Refactor**
	- Updated `HnippTemplate` class for future legacy support adjustments.
- **Deprecations**
	- Marked `Breadcrumbs`, `Button`, and `Icon` components as deprecated, signaling future removal and encouraging the use of updated components.
	- Deprecated `IconTypesDeprecated` type in favor of `IconTypes`.
- **Chores**
	- Removed the import of `React` in `Icon.tsx`. Added a deprecation comment for the `Icon` component.

### CODE REVIEW

**Review by rafnarnason — APPROVED** [review]
lgtm

File: libs/island-ui/core/src/lib/Icon/Icon.tsx [hunk 1]
```diff
@@ -274,6 +273,9 @@ const SvgPathContainer = ({
   )
 }
 
+/**
+ * @deprecated Use something else?
```
Comment (oddsson) [inline libs/island-ui/core/src/lib/Icon/Icon.tsx:277]: There is a newer Icon component, IconRC. We should point people there.

```suggestion
 * @deprecated Use IconRC instead
```

**Review by oddsson — APPROVED** [review]

**Comment by andes-it** [comment]
This PR's had a 15-day siesta. Lost its spark, did it? It's making an exit in 5 days unless the stale label disappears.

**Comment by andes-it** [comment]
This PR closed its doors. Sat around for a day after the stale tag, then reckoned it's not worth the fuss.

**Pull request closed without merging**